cmd/grype-db/cli/commands/cache_backup.go

```diff
@@ -138,13 +141,25 @@ func archiveProvider(root string, name string, writer *tar.Writer) error {
 
 	return filepath.Walk(name,
 		func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-
 			if info.IsDir() {
 				return nil
 			}
+			if strings.HasSuffix(path, "metadata.json") && cfg.Results.ResultsOnly {
```

Comment:
this will process any path that ends in `metadata.json`, and though this works today, we are really only intending to process `./metadata.json` relative to the root of the workspace.